### PR TITLE
Checkin of python-lint.yml

### PR DESCRIPTION
**Pull request by dave3d**
Adapted from python-app.yml Github Action file.  That file sets up
an action that runs flake8 and pytest on any pull request.

For this one, the pytest call is removed, and flake8 is only run
on the Examples sub-directory.

### CODE REVIEW

File: .github/workflows/python-lint.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```
Comment (blowekamp) [inline .github/workflows/python-lint.yml:27]: This line is not needed.

Comment (dave3d) [reply .github/workflows/python-lint.yml:27]: Oh, right.  I remember thinking that.  The script was autogenerated by Github.

**Comment by blowekamp** [comment]
Any idea why this didn't trigger the GH Action? It would be great to check it works before we merge.

Should a root .flake8 file be added for options like line length?

```
 # stop the build if there are Python syntax errors or undefined names
 flake8 Examples --count --select=E9,F63,F7,F82 --show-source --statistics
 # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
 flake8 Examples --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
```

This "select" is from the example, any idea if we should consider more problems errors to be more strict because if there is only a warning, the CI will not be read, and it won't be check. What do you think?

File: .github/workflows/python-lint.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
```
Comment (zivy) [inline .github/workflows/python-lint.yml:26]: Is pytest needed here or can it be removed?

Comment (dave3d) [reply .github/workflows/python-lint.yml:26]: Yeah, you're right.  Not needed.

**Pull request merged**